### PR TITLE
Create suggestion repository methods

### PR DESCRIPTION
**Pull request by tosigame**
This pull request adds essential methods for managing suggestions within trips and introduces tests that cover basic scenarios for each method. The core functionalities implemented are:

addSuggestionToTrip: Facilitates the addition of new suggestions to a trip.
updateSuggestionInTrip: Enables updating details of an existing suggestion.
removeSuggestionFromTrip: Allows for the removal of a suggestion from a trip.
getSuggestionFromTrip and getAllSuggestionsFromTrip: Provides mechanisms to fetch suggestions, either a specific one or all associated with a trip.
The accompanying tests ensure these methods perform correctly under straightforward conditions, laying the groundwork for future, more complex test scenarios.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt [hunk 1]
```diff
@@ -61,6 +63,204 @@ class TripsRepository(
...
+   * @param suggestion The `Suggestion` object to be added to the trip.
+   * @return `true` if the suggestion was added successfully, `false` otherwise. Errors during the
+   *   process are logged.
+   */
+  suspend fun addSuggestionToTrip(tripId: String, suggestion: Suggestion): Boolean =
+      withContext(dispatcher) {
+        try {
+          val uniqueID = UUID.randomUUID().toString()
+          val firebaseSuggestion =
+              FirestoreSuggestion.fromSuggestion(suggestion.copy(suggestionId = uniqueID))
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt:151]: TripsRepository already knows who is creating the Suggestion, it should also modify the userId and username of the Suggestion when copying the suggestion.

**Review by AgeX21 — CHANGES_REQUESTED** [review]
Everything seems fine except when adding the Suggestion to the Trip, just modify this part and it will be okay for me.

File: app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt (hunk 1 above)
Comment (tosigame) [reply app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt:151]: Great catch about the UserId, however for the username, this is something that the app, should know. As such the username will need to be specified by the caller. Thx!

File: app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt [hunk 2]
```diff
@@ -61,6 +63,207 @@ class TripsRepository(
...
+   */
+  suspend fun addSuggestionToTrip(tripId: String, suggestion: Suggestion): Boolean =
+      withContext(dispatcher) {
+        try {
+          val uniqueID = UUID.randomUUID().toString()
+          val firebaseSuggestion =
+              FirestoreSuggestion.fromSuggestion(
+                  suggestion.copy(
+                      suggestionId = uniqueID,
+                      userId = uid)) // we already know who creates the suggestion
```
Comment (AgeX21) [inline app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt:154]: Perfect.

**Review by AgeX21 — APPROVED** [review]
Looks good to me.

**Pull request merged**